lib/pacemaker/pcmk_sched_instances.c

```diff
@@ -1097,36 +1097,34 @@ update_noninterleaved_actions(pe_resource_t *instance, pe_action_t *first,
  * \return Group of enum pcmk__updated flags indicating what was updated
  */
 uint32_t
-pcmk__multi_update_actions(pe_action_t *first, pe_action_t *then,
-                           const pe_node_t *node, uint32_t flags,
-                           uint32_t filter, uint32_t type,
-                           pe_working_set_t *data_set)
+pcmk__instance_update_ordered_actions(pe_action_t *first, pe_action_t *then,
+                                      const pe_node_t *node, uint32_t flags,
+                                      uint32_t filter, uint32_t type,
+                                      pe_working_set_t *data_set)
 {
-    uint32_t changed = pcmk__updated_none;
-
-    crm_trace("%s -> %s", first->uuid, then->uuid);
+    if (then->rsc == NULL) {
+        return pcmk__updated_none;
 
-    if(can_interleave_actions(first, then)) {
-        changed = update_interleaved_actions(first, then, node, filter, type);
+    } else if (can_interleave_actions(first, then)) {
+        return update_interleaved_actions(first, then, node, filter, type);
 
-    } else if(then->rsc) {
-        GList *gIter = NULL;
-        GList *children = NULL;
+    } else {
+        uint32_t changed = pcmk__updated_none;
+        GList *instances = get_instance_list(then->rsc);
 
-        // Handle the 'primitive' ordering case
+        // Update actions for the clone or bundle resource itself
```

Comment:
Yeah it means call the primitive method for the clone or bundle. That's why (a while back) I renamed that function  `pcmk__update_ordered_actions()` instead of `pcmk__primitive_update_ordered_actions()`.